rescue/benches/rescue.rs

```diff
@@ -2,37 +2,40 @@ use std::any::type_name;
 
 use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
 use p3_baby_bear::BabyBear;
-use p3_field::PrimeField64;
+use p3_field::{AbstractField, Field, PrimeField64};
 use p3_goldilocks::Goldilocks;
 use p3_mds::babybear::MdsMatrixBabyBear;
 use p3_mds::goldilocks::MdsMatrixGoldilocks;
+use p3_mds::integrated_coset_mds::IntegratedCosetMds;
 use p3_mds::mersenne31::MdsMatrixMersenne31;
 use p3_mds::MdsPermutation;
 use p3_mersenne_31::Mersenne31;
-use p3_rescue::{BasicInverseSboxLayer, InverseSboxLayer, Rescue};
+use p3_rescue::{BasicSboxLayer, Rescue, SboxLayers};
 use p3_symmetric::permutation::CryptographicPermutation;
 use rand::distributions::{Distribution, Standard};
 use rand::{thread_rng, Rng};
 
 fn bench_rescue(c: &mut Criterion) {
-    rescue::<BabyBear, MdsMatrixBabyBear, BasicInverseSboxLayer, 16, 7>(c);
-    rescue::<BabyBear, MdsMatrixBabyBear, BasicInverseSboxLayer, 24, 7>(c);
-    rescue::<BabyBear, MdsMatrixBabyBear, BasicInverseSboxLayer, 32, 7>(c);
+    rescue::<BabyBear, IntegratedCosetMds<_, 16>, BasicSboxLayer<_>, 16, 7>(c);
+    rescue::<<BabyBear as Field>::Packing, IntegratedCosetMds<_, 16>, BasicSboxLayer<_>, 16, 7>(c);
+    rescue::<BabyBear, MdsMatrixBabyBear, BasicSboxLayer<_>, 24, 7>(c);
+    rescue::<BabyBear, MdsMatrixBabyBear, BasicSboxLayer<_>, 32, 7>(c);
 
-    rescue::<Goldilocks, MdsMatrixGoldilocks, BasicInverseSboxLayer, 8, 7>(c);
-    rescue::<Goldilocks, MdsMatrixGoldilocks, BasicInverseSboxLayer, 12, 7>(c);
-    rescue::<Goldilocks, MdsMatrixGoldilocks, BasicInverseSboxLayer, 16, 7>(c);
+    rescue::<Goldilocks, MdsMatrixGoldilocks, BasicSboxLayer<_>, 8, 7>(c);
+    rescue::<Goldilocks, MdsMatrixGoldilocks, BasicSboxLayer<_>, 12, 7>(c);
+    rescue::<Goldilocks, MdsMatrixGoldilocks, BasicSboxLayer<_>, 16, 7>(c);
 
-    rescue::<Mersenne31, MdsMatrixMersenne31, BasicInverseSboxLayer, 16, 5>(c);
-    rescue::<Mersenne31, MdsMatrixMersenne31, BasicInverseSboxLayer, 32, 5>(c);
+    rescue::<Mersenne31, MdsMatrixMersenne31, BasicSboxLayer<_>, 16, 5>(c);
+    rescue::<Mersenne31, MdsMatrixMersenne31, BasicSboxLayer<_>, 32, 5>(c);
 }
 
-fn rescue<F, Mds, Isl, const WIDTH: usize, const ALPHA: u64>(c: &mut Criterion)
+fn rescue<F, Mds, Sbox, const WIDTH: usize, const ALPHA: u64>(c: &mut Criterion)
 where
-    F: PrimeField64,
-    Standard: Distribution<F>,
+    F: AbstractField,
+    F::F: PrimeField64,
```

Comment:
Not a big deal, this `F::F: ` syntax is a little confusing... one option is changing the AbstractField name to `AF`, and another is changing AbstractField's `F` associated type to `UnderlyingField` (or `Field`)? But if you think it's best as is that's fine too

Reply:
I agree it looks odd... I think we should keep it short (since it comes up all over) so maybe `AF::F` is good, or a single word like `AF::Field` or `AF::Concrete`

Reply:
Went with `AF::F` for now